chore: build for production by default

tltv opened the pull request:
Removes need to add `-Pvaadin.productionMode` explicitly when building for production to make build more consistent with Maven starters. `productionMode` property is activated with `bootJar` and `bootBuildImage` tasks.

Fixes: #293

tltv committed:
chore: build for production by default

Removes need to add -Pvaadin.productionMode explicitly when building for production to make build more consistent with Maven starters. productionMode property is activated with bootJar and bootBuildImage tasks.

Fixes: #293

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/vaadin/base-starter-spring-gradle?pullRequest=294) <br/>All committers have signed the CLA.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/vaadin/base-starter-spring-gradle?pullRequest=294) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/vaadin/base-starter-spring-gradle?pullRequest=294) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/vaadin/base-starter-spring-gradle?pullRequest=294) it.</sub>

mstahv commented:
Nice progress 👍 Functionality is king, but I hope the build config can still be made such that it looks simple by default (production mode flag could implictly have this default and getting rid of exclusions).

tltv commented:
> production mode flag could implictly have this default

This is potentially something that Flow gradle plugin can handle.

tltv commented:
> This is potentially something that Flow gradle plugin can handle.

Prepared https://github.com/vaadin/flow/pull/22704. If that goes in, then we should be able to simplify this starter by removing whole productionMode activation since it's in the Flow Gradle plugin.

tltv committed:
bump to Vaadin 25.0.0-beta5

Simplified build.gradle since Vaadin Gradle plugin sets production mode automatically when run with bootJar or bootBuildImage task.
